go/chat/boxer.go

```diff
@@ -77,13 +83,27 @@ func (b *Boxer) makeErrorMessage(msg chat1.MessageBoxed, err UnboxingError) chat
 	})
 }
 
-// UnboxMessage unboxes a chat1.MessageBoxed into a keybase1.Message.  It finds
-// the appropriate keybase1.CryptKey.
-// The first return value is unusable if the err != nil
-// Returns (_, err) for non-permanent errors, and (MessageUnboxedError, nil) for permanent errors.
-// Permanent errors can be cached and must be treated as a value to deal with.
-// Whereas temporary errors are transient failures.
-func (b *Boxer) UnboxMessage(ctx context.Context, boxed chat1.MessageBoxed, finalizeInfo *chat1.ConversationFinalizeInfo) (chat1.MessageUnboxed, UnboxingError) {
+// UnboxMessage unboxes a chat1.MessageBoxed into a keybase1.Message. It finds
+// the appropriate keybase1.CryptKey, decrypts the message, and verifies
+// several things:
+//   - The message's signature is valid.
+//   - (TODO) The signing KID was valid when the signature was made.
```

Comment:
Yeah. `ctime` is untrusted and from the server.
In the case where the server tells the truth, we fail to unbox if the device was revoked before sending.
In the case where the server is lying about `ctime`, we successfully unbox even if the message was sent after its key was revoked (because we don't know that). But we set `MessageUnboxedValid.SenderDeviceRevokedAt` if the key was ever revoked. Which is supposed to show a warning on the gui.